WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift

```diff
@@ -297,20 +291,40 @@ private extension JetpackSetupCoordinator {
     }
 
     @MainActor
-    func fetchJetpackUser() async -> Result<JetpackUser, Error> {
+    func fetchJetpackUser() async throws -> JetpackUser {
         /// Jetpack setup will fail anyway without admin role, so check that first.
         let roles = stores.sessionManager.defaultRoles
         guard roles.contains(.administrator) else {
-            return .failure(JetpackCheckError.missingPermission)
+            throw JetpackCheckError.missingPermission
         }
-        return await withCheckedContinuation { continuation in
+        return try await withCheckedThrowingContinuation { continuation in
             let action = JetpackConnectionAction.fetchJetpackUser { result in
-                continuation.resume(returning: result)
+                continuation.resume(with: result)
             }
             stores.dispatch(action)
         }
     }
 
+    @MainActor
+    func isJetpackInstalledAndActive() async -> Bool {
+        await withCheckedContinuation { continuation in
+            stores.dispatch(SystemStatusAction.synchronizeSystemPlugins(siteID: 0) { result in
+                switch result {
+                case let .success(plugins):
+                    if let plugin = plugins.first(where: { $0.name.lowercased() == Constants.jetpackPluginName.lowercased() }),
+                       plugin.active {
+                        continuation.resume(returning: true)
+                    } else {
+                        continuation.resume(returning: false)
+                    }
+                case let .failure(error):
+                    DDLogError("⛔️ Failed to sync system plugins. Error: \(error)")
+                    continuation.resume(returning: false)
```

Comment:
❓ Should we display an alert for the error and block the user from proceeding?

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/11057/commits/0f15ea35c5b571a2835cb6dd4301f9a427818c32